core/src/main/java/io/questdb/griffin/engine/functions/conditional/NullIfStrFunctionFactory.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.conditional;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.StrFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.ObjList;
+
+public class NullIfStrFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "nullif(SS)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) {
+        Function strFunc1 = args.getQuick(0);
+        Function strFunc2 = args.getQuick(1);
+        return new Func(strFunc1, strFunc2);
+    }
+
+    private static class Func extends StrFunction implements BinaryFunction {
+        private final Function strFunc1;
+        private final Function strFunc2;
+
+        public Func(Function strFunc1, Function strFunc2) {
+            this.strFunc1 = strFunc1;
+            this.strFunc2 = strFunc2;
+        }
+
+        @Override
+        public Function getLeft() {
+            return strFunc1;
+        }
+
+        @Override
+        public Function getRight() {
+            return strFunc2;
+        }
+
+        @Override
+        public CharSequence getStr(Record rec) {
+            CharSequence cs1 = strFunc1.getStr(rec);
+            if (cs1 == null) {
+                return null;
+            }
+            CharSequence cs2 = strFunc2.getStr(rec);
+            if (cs2 == null) {
+                return null;
+            }
+            return cs1.equals(cs2) ? null : cs1;
```

Comment:
I think `Chars.equals` must be used here. `CharSequence.equals` is reference comparison, is't it?

Reply:
![image](https://user-images.githubusercontent.com/2159629/141165422-8375d3e0-f174-48d5-9717-20b28d14ed68.png)

Reply:
Nice catch! I overlooked that.

Reply:
To be precise, the current code is correct for us since both `AbstractCharSequence` and `String` override `equals` method. Nevertheless, I'm going to update the code to make things more straight-forward.

Reply:
Fixed in 0317f58a0ad330e31c9a163097bc6040a8a6ee92. Thanks for spotting the problem once again.